// @aws-sdk/client-opensearchserverless/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  AccessPolicyType: () => AccessPolicyType,
  BatchGetCollectionCommand: () => BatchGetCollectionCommand,
  BatchGetEffectiveLifecyclePolicyCommand: () => BatchGetEffectiveLifecyclePolicyCommand,
  BatchGetLifecyclePolicyCommand: () => BatchGetLifecyclePolicyCommand,
  BatchGetVpcEndpointCommand: () => BatchGetVpcEndpointCommand,
  CollectionStatus: () => CollectionStatus,
  CollectionType: () => CollectionType,
  ConflictException: () => ConflictException,
  CreateAccessPolicyCommand: () => CreateAccessPolicyCommand,
  CreateCollectionCommand: () => CreateCollectionCommand,
  CreateIndexCommand: () => CreateIndexCommand,
  CreateLifecyclePolicyCommand: () => CreateLifecyclePolicyCommand,
  CreateSecurityConfigCommand: () => CreateSecurityConfigCommand,
  CreateSecurityPolicyCommand: () => CreateSecurityPolicyCommand,
  CreateVpcEndpointCommand: () => CreateVpcEndpointCommand,
  DeleteAccessPolicyCommand: () => DeleteAccessPolicyCommand,
  DeleteCollectionCommand: () => DeleteCollectionCommand,
  DeleteIndexCommand: () => DeleteIndexCommand,
  DeleteLifecyclePolicyCommand: () => DeleteLifecyclePolicyCommand,
  DeleteSecurityConfigCommand: () => DeleteSecurityConfigCommand,
  DeleteSecurityPolicyCommand: () => DeleteSecurityPolicyCommand,
  DeleteVpcEndpointCommand: () => DeleteVpcEndpointCommand,
  GetAccessPolicyCommand: () => GetAccessPolicyCommand,
  GetAccountSettingsCommand: () => GetAccountSettingsCommand,
  GetIndexCommand: () => GetIndexCommand,
  GetPoliciesStatsCommand: () => GetPoliciesStatsCommand,
  GetSecurityConfigCommand: () => GetSecurityConfigCommand,
  GetSecurityPolicyCommand: () => GetSecurityPolicyCommand,
  IamIdentityCenterGroupAttribute: () => IamIdentityCenterGroupAttribute,
  IamIdentityCenterUserAttribute: () => IamIdentityCenterUserAttribute,
  InternalServerException: () => InternalServerException,
  LifecyclePolicyType: () => LifecyclePolicyType,
  ListAccessPoliciesCommand: () => ListAccessPoliciesCommand,
  ListCollectionsCommand: () => ListCollectionsCommand,
  ListLifecyclePoliciesCommand: () => ListLifecyclePoliciesCommand,
  ListSecurityConfigsCommand: () => ListSecurityConfigsCommand,
  ListSecurityPoliciesCommand: () => ListSecurityPoliciesCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListVpcEndpointsCommand: () => ListVpcEndpointsCommand,
  OcuLimitExceededException: () => OcuLimitExceededException,
  OpenSearchServerless: () => OpenSearchServerless,
  OpenSearchServerlessClient: () => OpenSearchServerlessClient,
  OpenSearchServerlessServiceException: () => OpenSearchServerlessServiceException,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ResourceType: () => ResourceType,
  SecurityConfigType: () => SecurityConfigType,
  SecurityPolicyType: () => SecurityPolicyType,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  StandbyReplicas: () => StandbyReplicas,
  TagResourceCommand: () => TagResourceCommand,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateAccessPolicyCommand: () => UpdateAccessPolicyCommand,
  UpdateAccountSettingsCommand: () => UpdateAccountSettingsCommand,
  UpdateCollectionCommand: () => UpdateCollectionCommand,
  UpdateIndexCommand: () => UpdateIndexCommand,
  UpdateLifecyclePolicyCommand: () => UpdateLifecyclePolicyCommand,
  UpdateSecurityConfigCommand: () => UpdateSecurityConfigCommand,
  UpdateSecurityPolicyCommand: () => UpdateSecurityPolicyCommand,
  UpdateVpcEndpointCommand: () => UpdateVpcEndpointCommand,
  ValidationException: () => ValidationException,
  VpcEndpointStatus: () => VpcEndpointStatus,
  __Client: () => import_smithy_client.Client,
  paginateListAccessPolicies: () => paginateListAccessPolicies,
  paginateListCollections: () => paginateListCollections,
  paginateListLifecyclePolicies: () => paginateListLifecyclePolicies,
  paginateListSecurityConfigs: () => paginateListSecurityConfigs,
  paginateListSecurityPolicies: () => paginateListSecurityPolicies,
  paginateListVpcEndpoints: () => paginateListVpcEndpoints
});
module.exports = __toCommonJS(index_exports);

// src/OpenSearchServerlessClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return Object.assign(options, {
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "aoss"
  });
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/OpenSearchServerlessClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = Object.assign(
    (0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig),
    (0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig),
    (0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig),
    getHttpAuthExtensionConfiguration(runtimeConfig)
  );
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return Object.assign(
    runtimeConfig,
    (0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    (0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    (0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    resolveHttpAuthRuntimeConfig(extensionConfiguration)
  );
}, "resolveRuntimeExtensions");

// src/OpenSearchServerlessClient.ts
var OpenSearchServerlessClient = class extends import_smithy_client.Client {
  static {
    __name(this, "OpenSearchServerlessClient");
  }
  /**
   * The resolved configuration of OpenSearchServerlessClient class. This is resolved and normalized from the {@link OpenSearchServerlessClientConfig | constructor configuration interface}.
   */
  config;
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    super(_config_0);
    this.initConfig = _config_0;
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_1);
    const _config_3 = (0, import_middleware_retry.resolveRetryConfig)(_config_2);
    const _config_4 = (0, import_config_resolver.resolveRegionConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: import_httpAuthSchemeProvider.defaultOpenSearchServerlessHttpAuthSchemeParametersProvider,
        identityProviderConfigProvider: /* @__PURE__ */ __name(async (config) => new import_core.DefaultIdentityProviderConfig({
          "aws.auth#sigv4": config.credentials
        }), "identityProviderConfigProvider")
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
};

// src/OpenSearchServerless.ts


// src/commands/BatchGetCollectionCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");


// src/protocols/Aws_json1_0.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");

// src/models/OpenSearchServerlessServiceException.ts

var OpenSearchServerlessServiceException = class _OpenSearchServerlessServiceException extends import_smithy_client.ServiceException {
  static {
    __name(this, "OpenSearchServerlessServiceException");
  }
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _OpenSearchServerlessServiceException.prototype);
  }
};

// src/models/models_0.ts
var AccessPolicyType = {
  /**
   * data policy type
   */
  data: "data"
};
var ConflictException = class _ConflictException extends OpenSearchServerlessServiceException {
  static {
    __name(this, "ConflictException");
  }
  name = "ConflictException";
  $fault = "client";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _ConflictException.prototype);
  }
};
var InternalServerException = class _InternalServerException extends OpenSearchServerlessServiceException {
  static {
    __name(this, "InternalServerException");
  }
  name = "InternalServerException";
  $fault = "server";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    Object.setPrototypeOf(this, _InternalServerException.prototype);
  }
};
var ServiceQuotaExceededException = class _ServiceQuotaExceededException extends OpenSearchServerlessServiceException {
  static {
    __name(this, "ServiceQuotaExceededException");
  }
  name = "ServiceQuotaExceededException";
  $fault = "client";
  /**
   * Identifier of the resource affected.
   * @public
   */
  resourceId;
  /**
   * Type of the resource affected.
   * @public
   */
  resourceType;
  /**
   * Service Quotas requirement to identify originating service.
   * @public
   */
  serviceCode;
  /**
   * Service Quotas requirement to identify originating quota.
   * @public
   */
  quotaCode;
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
    this.resourceId = opts.resourceId;
    this.resourceType = opts.resourceType;
    this.serviceCode = opts.serviceCode;
    this.quotaCode = opts.quotaCode;
  }
};
var ValidationException = class _ValidationException extends OpenSearchServerlessServiceException {
  static {
    __name(this, "ValidationException");
  }
  name = "ValidationException";
  $fault = "client";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _ValidationException.prototype);
  }
};
var ResourceNotFoundException = class _ResourceNotFoundException extends OpenSearchServerlessServiceException {
  static {
    __name(this, "ResourceNotFoundException");
  }
  name = "ResourceNotFoundException";
  $fault = "client";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
var StandbyReplicas = {
  /**
   * Standby replicas disabled
   */
  DISABLED: "DISABLED",
  /**
   * Standby replicas enabled
   */
  ENABLED: "ENABLED"
};
var CollectionStatus = {
  /**
   * Collection resource is ready to use
   */
  ACTIVE: "ACTIVE",
  /**
   * Creating collection resource
   */
  CREATING: "CREATING",
  /**
   * Deleting collection resource
   */
  DELETING: "DELETING",
  /**
   * Collection resource create or delete failed
   */
  FAILED: "FAILED"
};
var CollectionType = {
  /**
   * Search collection type
   */
  SEARCH: "SEARCH",
  /**
   * Timeseries collection type
   */
  TIMESERIES: "TIMESERIES",
  /**
   * Vectorsearch collection type
   */
  VECTORSEARCH: "VECTORSEARCH"
};
var LifecyclePolicyType = {
  /**
   * retention policy type
   */
  retention: "retention"
};
var ResourceType = {
  /**
   * index resource type
   */
  index: "index"
};
var VpcEndpointStatus = {
  /**
   * VPCEndpoint resource is ready to use
   */
  ACTIVE: "ACTIVE",
  /**
   * Deleting VPCEndpoint resource
   */
  DELETING: "DELETING",
  /**
   * VPCEndpoint resource create or delete failed
   */
  FAILED: "FAILED",
  /**
   * Pending VPCEndpoint resource
   */
  PENDING: "PENDING"
};
var OcuLimitExceededException = class _OcuLimitExceededException extends OpenSearchServerlessServiceException {
  static {
    __name(this, "OcuLimitExceededException");
  }
  name = "OcuLimitExceededException";
  $fault = "client";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "OcuLimitExceededException",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _OcuLimitExceededException.prototype);
  }
};
var IamIdentityCenterGroupAttribute = {
  /**
   * Group ID
   */
  GroupId: "GroupId",
  /**
   * Group Name
   */
  GroupName: "GroupName"
};
var IamIdentityCenterUserAttribute = {
  /**
   * Email
   */
  Email: "Email",
  /**
   * User ID
   */
  UserId: "UserId",
  /**
   * User Name
   */
  UserName: "UserName"
};
var SecurityConfigType = {
  /**
   * iam federation
   */
  iamfederation: "iamfederation",
  /**
   * iam identity center
   */
  iamidentitycenter: "iamidentitycenter",
  /**
   * saml provider
   */
  saml: "saml"
};
var SecurityPolicyType = {
  /**
   * encryption policy type
   */
  encryption: "encryption",
  /**
   * network policy type
   */
  network: "network"
};

// src/protocols/Aws_json1_0.ts
var se_BatchGetCollectionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("BatchGetCollection");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_BatchGetCollectionCommand");
var se_BatchGetEffectiveLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("BatchGetEffectiveLifecyclePolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_BatchGetEffectiveLifecyclePolicyCommand");
var se_BatchGetLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("BatchGetLifecyclePolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_BatchGetLifecyclePolicyCommand");
var se_BatchGetVpcEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("BatchGetVpcEndpoint");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_BatchGetVpcEndpointCommand");
var se_CreateAccessPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateAccessPolicy");
  let body;
  body = JSON.stringify(se_CreateAccessPolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateAccessPolicyCommand");
var se_CreateCollectionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCollection");
  let body;
  body = JSON.stringify(se_CreateCollectionRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateCollectionCommand");
var se_CreateIndexCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateIndex");
  let body;
  body = JSON.stringify(se_CreateIndexRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateIndexCommand");
var se_CreateLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateLifecyclePolicy");
  let body;
  body = JSON.stringify(se_CreateLifecyclePolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateLifecyclePolicyCommand");
var se_CreateSecurityConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateSecurityConfig");
  let body;
  body = JSON.stringify(se_CreateSecurityConfigRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateSecurityConfigCommand");
var se_CreateSecurityPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateSecurityPolicy");
  let body;
  body = JSON.stringify(se_CreateSecurityPolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateSecurityPolicyCommand");
var se_CreateVpcEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateVpcEndpoint");
  let body;
  body = JSON.stringify(se_CreateVpcEndpointRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateVpcEndpointCommand");
var se_DeleteAccessPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAccessPolicy");
  let body;
  body = JSON.stringify(se_DeleteAccessPolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAccessPolicyCommand");
var se_DeleteCollectionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCollection");
  let body;
  body = JSON.stringify(se_DeleteCollectionRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCollectionCommand");
var se_DeleteIndexCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteIndex");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteIndexCommand");
var se_DeleteLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteLifecyclePolicy");
  let body;
  body = JSON.stringify(se_DeleteLifecyclePolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteLifecyclePolicyCommand");
var se_DeleteSecurityConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteSecurityConfig");
  let body;
  body = JSON.stringify(se_DeleteSecurityConfigRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteSecurityConfigCommand");
var se_DeleteSecurityPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteSecurityPolicy");
  let body;
  body = JSON.stringify(se_DeleteSecurityPolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteSecurityPolicyCommand");
var se_DeleteVpcEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteVpcEndpoint");
  let body;
  body = JSON.stringify(se_DeleteVpcEndpointRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteVpcEndpointCommand");
var se_GetAccessPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAccessPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAccessPolicyCommand");
var se_GetAccountSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAccountSettings");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAccountSettingsCommand");
var se_GetIndexCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetIndex");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetIndexCommand");
var se_GetPoliciesStatsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetPoliciesStats");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetPoliciesStatsCommand");
var se_GetSecurityConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetSecurityConfig");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetSecurityConfigCommand");
var se_GetSecurityPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetSecurityPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetSecurityPolicyCommand");
var se_ListAccessPoliciesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAccessPolicies");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAccessPoliciesCommand");
var se_ListCollectionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCollections");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCollectionsCommand");
var se_ListLifecyclePoliciesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListLifecyclePolicies");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListLifecyclePoliciesCommand");
var se_ListSecurityConfigsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListSecurityConfigs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListSecurityConfigsCommand");
var se_ListSecurityPoliciesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListSecurityPolicies");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListSecurityPoliciesCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_ListVpcEndpointsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListVpcEndpoints");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListVpcEndpointsCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var se_UpdateAccessPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateAccessPolicy");
  let body;
  body = JSON.stringify(se_UpdateAccessPolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateAccessPolicyCommand");
var se_UpdateAccountSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateAccountSettings");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateAccountSettingsCommand");
var se_UpdateCollectionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateCollection");
  let body;
  body = JSON.stringify(se_UpdateCollectionRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateCollectionCommand");
var se_UpdateIndexCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateIndex");
  let body;
  body = JSON.stringify(se_UpdateIndexRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateIndexCommand");
var se_UpdateLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateLifecyclePolicy");
  let body;
  body = JSON.stringify(se_UpdateLifecyclePolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateLifecyclePolicyCommand");
var se_UpdateSecurityConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateSecurityConfig");
  let body;
  body = JSON.stringify(se_UpdateSecurityConfigRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateSecurityConfigCommand");
var se_UpdateSecurityPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateSecurityPolicy");
  let body;
  body = JSON.stringify(se_UpdateSecurityPolicyRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateSecurityPolicyCommand");
var se_UpdateVpcEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateVpcEndpoint");
  let body;
  body = JSON.stringify(se_UpdateVpcEndpointRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateVpcEndpointCommand");
var de_BatchGetCollectionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_BatchGetCollectionCommand");
var de_BatchGetEffectiveLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_BatchGetEffectiveLifecyclePolicyCommand");
var de_BatchGetLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_BatchGetLifecyclePolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_BatchGetLifecyclePolicyCommand");
var de_BatchGetVpcEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_BatchGetVpcEndpointCommand");
var de_CreateAccessPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateAccessPolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateAccessPolicyCommand");
var de_CreateCollectionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateCollectionCommand");
var de_CreateIndexCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateIndexCommand");
var de_CreateLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateLifecyclePolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateLifecyclePolicyCommand");
var de_CreateSecurityConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateSecurityConfigCommand");
var de_CreateSecurityPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateSecurityPolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateSecurityPolicyCommand");
var de_CreateVpcEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateVpcEndpointCommand");
var de_DeleteAccessPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteAccessPolicyCommand");
var de_DeleteCollectionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteCollectionCommand");
var de_DeleteIndexCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteIndexCommand");
var de_DeleteLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteLifecyclePolicyCommand");
var de_DeleteSecurityConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteSecurityConfigCommand");
var de_DeleteSecurityPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteSecurityPolicyCommand");
var de_DeleteVpcEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteVpcEndpointCommand");
var de_GetAccessPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetAccessPolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAccessPolicyCommand");
var de_GetAccountSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAccountSettingsCommand");
var de_GetIndexCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetIndexResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetIndexCommand");
var de_GetPoliciesStatsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetPoliciesStatsCommand");
var de_GetSecurityConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetSecurityConfigCommand");
var de_GetSecurityPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetSecurityPolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetSecurityPolicyCommand");
var de_ListAccessPoliciesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAccessPoliciesCommand");
var de_ListCollectionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCollectionsCommand");
var de_ListLifecyclePoliciesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListLifecyclePoliciesCommand");
var de_ListSecurityConfigsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListSecurityConfigsCommand");
var de_ListSecurityPoliciesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListSecurityPoliciesCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_ListVpcEndpointsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListVpcEndpointsCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_UpdateAccessPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateAccessPolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateAccessPolicyCommand");
var de_UpdateAccountSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateAccountSettingsCommand");
var de_UpdateCollectionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateCollectionCommand");
var de_UpdateIndexCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateIndexCommand");
var de_UpdateLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateLifecyclePolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateLifecyclePolicyCommand");
var de_UpdateSecurityConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateSecurityConfigCommand");
var de_UpdateSecurityPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateSecurityPolicyResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateSecurityPolicyCommand");
var de_UpdateVpcEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateVpcEndpointCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "InternalServerException":
    case "com.amazonaws.opensearchserverless#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.opensearchserverless#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.opensearchserverless#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.opensearchserverless#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    case "OcuLimitExceededException":
    case "com.amazonaws.opensearchserverless#OcuLimitExceededException":
      throw await de_OcuLimitExceededExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.opensearchserverless#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalServerExceptionRes");
var de_OcuLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new OcuLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_OcuLimitExceededExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ValidationExceptionRes");
var se_CreateAccessPolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    name: [],
    policy: [],
    type: []
  });
}, "se_CreateAccessPolicyRequest");
var se_CreateCollectionRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    name: [],
    standbyReplicas: [],
    tags: import_smithy_client._json,
    type: []
  });
}, "se_CreateCollectionRequest");
var se_CreateIndexRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    id: [],
    indexName: [],
    indexSchema: /* @__PURE__ */ __name((_) => se_IndexSchema(_, context), "indexSchema")
  });
}, "se_CreateIndexRequest");
var se_CreateLifecyclePolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    name: [],
    policy: [],
    type: []
  });
}, "se_CreateLifecyclePolicyRequest");
var se_CreateSecurityConfigRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    iamFederationOptions: import_smithy_client._json,
    iamIdentityCenterOptions: import_smithy_client._json,
    name: [],
    samlOptions: import_smithy_client._json,
    type: []
  });
}, "se_CreateSecurityConfigRequest");
var se_CreateSecurityPolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    name: [],
    policy: [],
    type: []
  });
}, "se_CreateSecurityPolicyRequest");
var se_CreateVpcEndpointRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    name: [],
    securityGroupIds: import_smithy_client._json,
    subnetIds: import_smithy_client._json,
    vpcId: []
  });
}, "se_CreateVpcEndpointRequest");
var se_DeleteAccessPolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    name: [],
    type: []
  });
}, "se_DeleteAccessPolicyRequest");
var se_DeleteCollectionRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    id: []
  });
}, "se_DeleteCollectionRequest");
var se_DeleteLifecyclePolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    name: [],
    type: []
  });
}, "se_DeleteLifecyclePolicyRequest");
var se_DeleteSecurityConfigRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    id: []
  });
}, "se_DeleteSecurityConfigRequest");
var se_DeleteSecurityPolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    name: [],
    type: []
  });
}, "se_DeleteSecurityPolicyRequest");
var se_DeleteVpcEndpointRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    id: []
  });
}, "se_DeleteVpcEndpointRequest");
var se_IndexSchema = /* @__PURE__ */ __name((input, context) => {
  return input;
}, "se_IndexSchema");
var se_UpdateAccessPolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    name: [],
    policy: [],
    policyVersion: [],
    type: []
  });
}, "se_UpdateAccessPolicyRequest");
var se_UpdateCollectionRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    id: []
  });
}, "se_UpdateCollectionRequest");
var se_UpdateIndexRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    id: [],
    indexName: [],
    indexSchema: /* @__PURE__ */ __name((_) => se_IndexSchema(_, context), "indexSchema")
  });
}, "se_UpdateIndexRequest");
var se_UpdateLifecyclePolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    name: [],
    policy: [],
    policyVersion: [],
    type: []
  });
}, "se_UpdateLifecyclePolicyRequest");
var se_UpdateSecurityConfigRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    configVersion: [],
    description: [],
    iamFederationOptions: import_smithy_client._json,
    iamIdentityCenterOptionsUpdates: import_smithy_client._json,
    id: [],
    samlOptions: import_smithy_client._json
  });
}, "se_UpdateSecurityConfigRequest");
var se_UpdateSecurityPolicyRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    description: [],
    name: [],
    policy: [],
    policyVersion: [],
    type: []
  });
}, "se_UpdateSecurityPolicyRequest");
var se_UpdateVpcEndpointRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    addSecurityGroupIds: import_smithy_client._json,
    addSubnetIds: import_smithy_client._json,
    clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    id: [],
    removeSecurityGroupIds: import_smithy_client._json,
    removeSubnetIds: import_smithy_client._json
  });
}, "se_UpdateVpcEndpointRequest");
var de_AccessPolicyDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    createdDate: import_smithy_client.expectLong,
    description: import_smithy_client.expectString,
    lastModifiedDate: import_smithy_client.expectLong,
    name: import_smithy_client.expectString,
    policy: /* @__PURE__ */ __name((_) => de_Document(_, context), "policy"),
    policyVersion: import_smithy_client.expectString,
    type: import_smithy_client.expectString
  });
}, "de_AccessPolicyDetail");
var de_BatchGetLifecyclePolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lifecyclePolicyDetails: /* @__PURE__ */ __name((_) => de_LifecyclePolicyDetails(_, context), "lifecyclePolicyDetails"),
    lifecyclePolicyErrorDetails: import_smithy_client._json
  });
}, "de_BatchGetLifecyclePolicyResponse");
var de_CreateAccessPolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accessPolicyDetail: /* @__PURE__ */ __name((_) => de_AccessPolicyDetail(_, context), "accessPolicyDetail")
  });
}, "de_CreateAccessPolicyResponse");
var de_CreateLifecyclePolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lifecyclePolicyDetail: /* @__PURE__ */ __name((_) => de_LifecyclePolicyDetail(_, context), "lifecyclePolicyDetail")
  });
}, "de_CreateLifecyclePolicyResponse");
var de_CreateSecurityPolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    securityPolicyDetail: /* @__PURE__ */ __name((_) => de_SecurityPolicyDetail(_, context), "securityPolicyDetail")
  });
}, "de_CreateSecurityPolicyResponse");
var de_GetAccessPolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accessPolicyDetail: /* @__PURE__ */ __name((_) => de_AccessPolicyDetail(_, context), "accessPolicyDetail")
  });
}, "de_GetAccessPolicyResponse");
var de_GetIndexResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    indexSchema: /* @__PURE__ */ __name((_) => de_IndexSchema(_, context), "indexSchema")
  });
}, "de_GetIndexResponse");
var de_GetSecurityPolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    securityPolicyDetail: /* @__PURE__ */ __name((_) => de_SecurityPolicyDetail(_, context), "securityPolicyDetail")
  });
}, "de_GetSecurityPolicyResponse");
var de_IndexSchema = /* @__PURE__ */ __name((output, context) => {
  return output;
}, "de_IndexSchema");
var de_LifecyclePolicyDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    createdDate: import_smithy_client.expectLong,
    description: import_smithy_client.expectString,
    lastModifiedDate: import_smithy_client.expectLong,
    name: import_smithy_client.expectString,
    policy: /* @__PURE__ */ __name((_) => de_Document(_, context), "policy"),
    policyVersion: import_smithy_client.expectString,
    type: import_smithy_client.expectString
  });
}, "de_LifecyclePolicyDetail");
var de_LifecyclePolicyDetails = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LifecyclePolicyDetail(entry, context);
  });
  return retVal;
}, "de_LifecyclePolicyDetails");
var de_SecurityPolicyDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    createdDate: import_smithy_client.expectLong,
    description: import_smithy_client.expectString,
    lastModifiedDate: import_smithy_client.expectLong,
    name: import_smithy_client.expectString,
    policy: /* @__PURE__ */ __name((_) => de_Document(_, context), "policy"),
    policyVersion: import_smithy_client.expectString,
    type: import_smithy_client.expectString
  });
}, "de_SecurityPolicyDetail");
var de_UpdateAccessPolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accessPolicyDetail: /* @__PURE__ */ __name((_) => de_AccessPolicyDetail(_, context), "accessPolicyDetail")
  });
}, "de_UpdateAccessPolicyResponse");
var de_UpdateLifecyclePolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lifecyclePolicyDetail: /* @__PURE__ */ __name((_) => de_LifecyclePolicyDetail(_, context), "lifecyclePolicyDetail")
  });
}, "de_UpdateLifecyclePolicyResponse");
var de_UpdateSecurityPolicyResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    securityPolicyDetail: /* @__PURE__ */ __name((_) => de_SecurityPolicyDetail(_, context), "securityPolicyDetail")
  });
}, "de_UpdateSecurityPolicyResponse");
var de_Document = /* @__PURE__ */ __name((output, context) => {
  return output;
}, "de_Document");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(OpenSearchServerlessServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.0",
    "x-amz-target": `OpenSearchServerless.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/BatchGetCollectionCommand.ts
var BatchGetCollectionCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "BatchGetCollection", {}).n("OpenSearchServerlessClient", "BatchGetCollectionCommand").f(void 0, void 0).ser(se_BatchGetCollectionCommand).de(de_BatchGetCollectionCommand).build() {
  static {
    __name(this, "BatchGetCollectionCommand");
  }
};

// src/commands/BatchGetEffectiveLifecyclePolicyCommand.ts



var BatchGetEffectiveLifecyclePolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "BatchGetEffectiveLifecyclePolicy", {}).n("OpenSearchServerlessClient", "BatchGetEffectiveLifecyclePolicyCommand").f(void 0, void 0).ser(se_BatchGetEffectiveLifecyclePolicyCommand).de(de_BatchGetEffectiveLifecyclePolicyCommand).build() {
  static {
    __name(this, "BatchGetEffectiveLifecyclePolicyCommand");
  }
};

// src/commands/BatchGetLifecyclePolicyCommand.ts



var BatchGetLifecyclePolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "BatchGetLifecyclePolicy", {}).n("OpenSearchServerlessClient", "BatchGetLifecyclePolicyCommand").f(void 0, void 0).ser(se_BatchGetLifecyclePolicyCommand).de(de_BatchGetLifecyclePolicyCommand).build() {
  static {
    __name(this, "BatchGetLifecyclePolicyCommand");
  }
};

// src/commands/BatchGetVpcEndpointCommand.ts



var BatchGetVpcEndpointCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "BatchGetVpcEndpoint", {}).n("OpenSearchServerlessClient", "BatchGetVpcEndpointCommand").f(void 0, void 0).ser(se_BatchGetVpcEndpointCommand).de(de_BatchGetVpcEndpointCommand).build() {
  static {
    __name(this, "BatchGetVpcEndpointCommand");
  }
};

// src/commands/CreateAccessPolicyCommand.ts



var CreateAccessPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "CreateAccessPolicy", {}).n("OpenSearchServerlessClient", "CreateAccessPolicyCommand").f(void 0, void 0).ser(se_CreateAccessPolicyCommand).de(de_CreateAccessPolicyCommand).build() {
  static {
    __name(this, "CreateAccessPolicyCommand");
  }
};

// src/commands/CreateCollectionCommand.ts



var CreateCollectionCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "CreateCollection", {}).n("OpenSearchServerlessClient", "CreateCollectionCommand").f(void 0, void 0).ser(se_CreateCollectionCommand).de(de_CreateCollectionCommand).build() {
  static {
    __name(this, "CreateCollectionCommand");
  }
};

// src/commands/CreateIndexCommand.ts



var CreateIndexCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "CreateIndex", {}).n("OpenSearchServerlessClient", "CreateIndexCommand").f(void 0, void 0).ser(se_CreateIndexCommand).de(de_CreateIndexCommand).build() {
  static {
    __name(this, "CreateIndexCommand");
  }
};

// src/commands/CreateLifecyclePolicyCommand.ts



var CreateLifecyclePolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "CreateLifecyclePolicy", {}).n("OpenSearchServerlessClient", "CreateLifecyclePolicyCommand").f(void 0, void 0).ser(se_CreateLifecyclePolicyCommand).de(de_CreateLifecyclePolicyCommand).build() {
  static {
    __name(this, "CreateLifecyclePolicyCommand");
  }
};

// src/commands/CreateSecurityConfigCommand.ts



var CreateSecurityConfigCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "CreateSecurityConfig", {}).n("OpenSearchServerlessClient", "CreateSecurityConfigCommand").f(void 0, void 0).ser(se_CreateSecurityConfigCommand).de(de_CreateSecurityConfigCommand).build() {
  static {
    __name(this, "CreateSecurityConfigCommand");
  }
};

// src/commands/CreateSecurityPolicyCommand.ts



var CreateSecurityPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "CreateSecurityPolicy", {}).n("OpenSearchServerlessClient", "CreateSecurityPolicyCommand").f(void 0, void 0).ser(se_CreateSecurityPolicyCommand).de(de_CreateSecurityPolicyCommand).build() {
  static {
    __name(this, "CreateSecurityPolicyCommand");
  }
};

// src/commands/CreateVpcEndpointCommand.ts



var CreateVpcEndpointCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "CreateVpcEndpoint", {}).n("OpenSearchServerlessClient", "CreateVpcEndpointCommand").f(void 0, void 0).ser(se_CreateVpcEndpointCommand).de(de_CreateVpcEndpointCommand).build() {
  static {
    __name(this, "CreateVpcEndpointCommand");
  }
};

// src/commands/DeleteAccessPolicyCommand.ts



var DeleteAccessPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "DeleteAccessPolicy", {}).n("OpenSearchServerlessClient", "DeleteAccessPolicyCommand").f(void 0, void 0).ser(se_DeleteAccessPolicyCommand).de(de_DeleteAccessPolicyCommand).build() {
  static {
    __name(this, "DeleteAccessPolicyCommand");
  }
};

// src/commands/DeleteCollectionCommand.ts



var DeleteCollectionCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "DeleteCollection", {}).n("OpenSearchServerlessClient", "DeleteCollectionCommand").f(void 0, void 0).ser(se_DeleteCollectionCommand).de(de_DeleteCollectionCommand).build() {
  static {
    __name(this, "DeleteCollectionCommand");
  }
};

// src/commands/DeleteIndexCommand.ts



var DeleteIndexCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "DeleteIndex", {}).n("OpenSearchServerlessClient", "DeleteIndexCommand").f(void 0, void 0).ser(se_DeleteIndexCommand).de(de_DeleteIndexCommand).build() {
  static {
    __name(this, "DeleteIndexCommand");
  }
};

// src/commands/DeleteLifecyclePolicyCommand.ts



var DeleteLifecyclePolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "DeleteLifecyclePolicy", {}).n("OpenSearchServerlessClient", "DeleteLifecyclePolicyCommand").f(void 0, void 0).ser(se_DeleteLifecyclePolicyCommand).de(de_DeleteLifecyclePolicyCommand).build() {
  static {
    __name(this, "DeleteLifecyclePolicyCommand");
  }
};

// src/commands/DeleteSecurityConfigCommand.ts



var DeleteSecurityConfigCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "DeleteSecurityConfig", {}).n("OpenSearchServerlessClient", "DeleteSecurityConfigCommand").f(void 0, void 0).ser(se_DeleteSecurityConfigCommand).de(de_DeleteSecurityConfigCommand).build() {
  static {
    __name(this, "DeleteSecurityConfigCommand");
  }
};

// src/commands/DeleteSecurityPolicyCommand.ts



var DeleteSecurityPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "DeleteSecurityPolicy", {}).n("OpenSearchServerlessClient", "DeleteSecurityPolicyCommand").f(void 0, void 0).ser(se_DeleteSecurityPolicyCommand).de(de_DeleteSecurityPolicyCommand).build() {
  static {
    __name(this, "DeleteSecurityPolicyCommand");
  }
};

// src/commands/DeleteVpcEndpointCommand.ts



var DeleteVpcEndpointCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "DeleteVpcEndpoint", {}).n("OpenSearchServerlessClient", "DeleteVpcEndpointCommand").f(void 0, void 0).ser(se_DeleteVpcEndpointCommand).de(de_DeleteVpcEndpointCommand).build() {
  static {
    __name(this, "DeleteVpcEndpointCommand");
  }
};

// src/commands/GetAccessPolicyCommand.ts



var GetAccessPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "GetAccessPolicy", {}).n("OpenSearchServerlessClient", "GetAccessPolicyCommand").f(void 0, void 0).ser(se_GetAccessPolicyCommand).de(de_GetAccessPolicyCommand).build() {
  static {
    __name(this, "GetAccessPolicyCommand");
  }
};

// src/commands/GetAccountSettingsCommand.ts



var GetAccountSettingsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "GetAccountSettings", {}).n("OpenSearchServerlessClient", "GetAccountSettingsCommand").f(void 0, void 0).ser(se_GetAccountSettingsCommand).de(de_GetAccountSettingsCommand).build() {
  static {
    __name(this, "GetAccountSettingsCommand");
  }
};

// src/commands/GetIndexCommand.ts



var GetIndexCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "GetIndex", {}).n("OpenSearchServerlessClient", "GetIndexCommand").f(void 0, void 0).ser(se_GetIndexCommand).de(de_GetIndexCommand).build() {
  static {
    __name(this, "GetIndexCommand");
  }
};

// src/commands/GetPoliciesStatsCommand.ts



var GetPoliciesStatsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "GetPoliciesStats", {}).n("OpenSearchServerlessClient", "GetPoliciesStatsCommand").f(void 0, void 0).ser(se_GetPoliciesStatsCommand).de(de_GetPoliciesStatsCommand).build() {
  static {
    __name(this, "GetPoliciesStatsCommand");
  }
};

// src/commands/GetSecurityConfigCommand.ts



var GetSecurityConfigCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "GetSecurityConfig", {}).n("OpenSearchServerlessClient", "GetSecurityConfigCommand").f(void 0, void 0).ser(se_GetSecurityConfigCommand).de(de_GetSecurityConfigCommand).build() {
  static {
    __name(this, "GetSecurityConfigCommand");
  }
};

// src/commands/GetSecurityPolicyCommand.ts



var GetSecurityPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "GetSecurityPolicy", {}).n("OpenSearchServerlessClient", "GetSecurityPolicyCommand").f(void 0, void 0).ser(se_GetSecurityPolicyCommand).de(de_GetSecurityPolicyCommand).build() {
  static {
    __name(this, "GetSecurityPolicyCommand");
  }
};

// src/commands/ListAccessPoliciesCommand.ts



var ListAccessPoliciesCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "ListAccessPolicies", {}).n("OpenSearchServerlessClient", "ListAccessPoliciesCommand").f(void 0, void 0).ser(se_ListAccessPoliciesCommand).de(de_ListAccessPoliciesCommand).build() {
  static {
    __name(this, "ListAccessPoliciesCommand");
  }
};

// src/commands/ListCollectionsCommand.ts



var ListCollectionsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "ListCollections", {}).n("OpenSearchServerlessClient", "ListCollectionsCommand").f(void 0, void 0).ser(se_ListCollectionsCommand).de(de_ListCollectionsCommand).build() {
  static {
    __name(this, "ListCollectionsCommand");
  }
};

// src/commands/ListLifecyclePoliciesCommand.ts



var ListLifecyclePoliciesCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "ListLifecyclePolicies", {}).n("OpenSearchServerlessClient", "ListLifecyclePoliciesCommand").f(void 0, void 0).ser(se_ListLifecyclePoliciesCommand).de(de_ListLifecyclePoliciesCommand).build() {
  static {
    __name(this, "ListLifecyclePoliciesCommand");
  }
};

// src/commands/ListSecurityConfigsCommand.ts



var ListSecurityConfigsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "ListSecurityConfigs", {}).n("OpenSearchServerlessClient", "ListSecurityConfigsCommand").f(void 0, void 0).ser(se_ListSecurityConfigsCommand).de(de_ListSecurityConfigsCommand).build() {
  static {
    __name(this, "ListSecurityConfigsCommand");
  }
};

// src/commands/ListSecurityPoliciesCommand.ts



var ListSecurityPoliciesCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "ListSecurityPolicies", {}).n("OpenSearchServerlessClient", "ListSecurityPoliciesCommand").f(void 0, void 0).ser(se_ListSecurityPoliciesCommand).de(de_ListSecurityPoliciesCommand).build() {
  static {
    __name(this, "ListSecurityPoliciesCommand");
  }
};

// src/commands/ListTagsForResourceCommand.ts



var ListTagsForResourceCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "ListTagsForResource", {}).n("OpenSearchServerlessClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
  static {
    __name(this, "ListTagsForResourceCommand");
  }
};

// src/commands/ListVpcEndpointsCommand.ts



var ListVpcEndpointsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "ListVpcEndpoints", {}).n("OpenSearchServerlessClient", "ListVpcEndpointsCommand").f(void 0, void 0).ser(se_ListVpcEndpointsCommand).de(de_ListVpcEndpointsCommand).build() {
  static {
    __name(this, "ListVpcEndpointsCommand");
  }
};

// src/commands/TagResourceCommand.ts



var TagResourceCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "TagResource", {}).n("OpenSearchServerlessClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
  static {
    __name(this, "TagResourceCommand");
  }
};

// src/commands/UntagResourceCommand.ts



var UntagResourceCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UntagResource", {}).n("OpenSearchServerlessClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
  static {
    __name(this, "UntagResourceCommand");
  }
};

// src/commands/UpdateAccessPolicyCommand.ts



var UpdateAccessPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateAccessPolicy", {}).n("OpenSearchServerlessClient", "UpdateAccessPolicyCommand").f(void 0, void 0).ser(se_UpdateAccessPolicyCommand).de(de_UpdateAccessPolicyCommand).build() {
  static {
    __name(this, "UpdateAccessPolicyCommand");
  }
};

// src/commands/UpdateAccountSettingsCommand.ts



var UpdateAccountSettingsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateAccountSettings", {}).n("OpenSearchServerlessClient", "UpdateAccountSettingsCommand").f(void 0, void 0).ser(se_UpdateAccountSettingsCommand).de(de_UpdateAccountSettingsCommand).build() {
  static {
    __name(this, "UpdateAccountSettingsCommand");
  }
};

// src/commands/UpdateCollectionCommand.ts



var UpdateCollectionCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateCollection", {}).n("OpenSearchServerlessClient", "UpdateCollectionCommand").f(void 0, void 0).ser(se_UpdateCollectionCommand).de(de_UpdateCollectionCommand).build() {
  static {
    __name(this, "UpdateCollectionCommand");
  }
};

// src/commands/UpdateIndexCommand.ts



var UpdateIndexCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateIndex", {}).n("OpenSearchServerlessClient", "UpdateIndexCommand").f(void 0, void 0).ser(se_UpdateIndexCommand).de(de_UpdateIndexCommand).build() {
  static {
    __name(this, "UpdateIndexCommand");
  }
};

// src/commands/UpdateLifecyclePolicyCommand.ts



var UpdateLifecyclePolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateLifecyclePolicy", {}).n("OpenSearchServerlessClient", "UpdateLifecyclePolicyCommand").f(void 0, void 0).ser(se_UpdateLifecyclePolicyCommand).de(de_UpdateLifecyclePolicyCommand).build() {
  static {
    __name(this, "UpdateLifecyclePolicyCommand");
  }
};

// src/commands/UpdateSecurityConfigCommand.ts



var UpdateSecurityConfigCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateSecurityConfig", {}).n("OpenSearchServerlessClient", "UpdateSecurityConfigCommand").f(void 0, void 0).ser(se_UpdateSecurityConfigCommand).de(de_UpdateSecurityConfigCommand).build() {
  static {
    __name(this, "UpdateSecurityConfigCommand");
  }
};

// src/commands/UpdateSecurityPolicyCommand.ts



var UpdateSecurityPolicyCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateSecurityPolicy", {}).n("OpenSearchServerlessClient", "UpdateSecurityPolicyCommand").f(void 0, void 0).ser(se_UpdateSecurityPolicyCommand).de(de_UpdateSecurityPolicyCommand).build() {
  static {
    __name(this, "UpdateSecurityPolicyCommand");
  }
};

// src/commands/UpdateVpcEndpointCommand.ts



var UpdateVpcEndpointCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("OpenSearchServerless", "UpdateVpcEndpoint", {}).n("OpenSearchServerlessClient", "UpdateVpcEndpointCommand").f(void 0, void 0).ser(se_UpdateVpcEndpointCommand).de(de_UpdateVpcEndpointCommand).build() {
  static {
    __name(this, "UpdateVpcEndpointCommand");
  }
};

// src/OpenSearchServerless.ts
var commands = {
  BatchGetCollectionCommand,
  BatchGetEffectiveLifecyclePolicyCommand,
  BatchGetLifecyclePolicyCommand,
  BatchGetVpcEndpointCommand,
  CreateAccessPolicyCommand,
  CreateCollectionCommand,
  CreateIndexCommand,
  CreateLifecyclePolicyCommand,
  CreateSecurityConfigCommand,
  CreateSecurityPolicyCommand,
  CreateVpcEndpointCommand,
  DeleteAccessPolicyCommand,
  DeleteCollectionCommand,
  DeleteIndexCommand,
  DeleteLifecyclePolicyCommand,
  DeleteSecurityConfigCommand,
  DeleteSecurityPolicyCommand,
  DeleteVpcEndpointCommand,
  GetAccessPolicyCommand,
  GetAccountSettingsCommand,
  GetIndexCommand,
  GetPoliciesStatsCommand,
  GetSecurityConfigCommand,
  GetSecurityPolicyCommand,
  ListAccessPoliciesCommand,
  ListCollectionsCommand,
  ListLifecyclePoliciesCommand,
  ListSecurityConfigsCommand,
  ListSecurityPoliciesCommand,
  ListTagsForResourceCommand,
  ListVpcEndpointsCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAccessPolicyCommand,
  UpdateAccountSettingsCommand,
  UpdateCollectionCommand,
  UpdateIndexCommand,
  UpdateLifecyclePolicyCommand,
  UpdateSecurityConfigCommand,
  UpdateSecurityPolicyCommand,
  UpdateVpcEndpointCommand
};
var OpenSearchServerless = class extends OpenSearchServerlessClient {
  static {
    __name(this, "OpenSearchServerless");
  }
};
(0, import_smithy_client.createAggregatedClient)(commands, OpenSearchServerless);

// src/pagination/ListAccessPoliciesPaginator.ts

var paginateListAccessPolicies = (0, import_core.createPaginator)(OpenSearchServerlessClient, ListAccessPoliciesCommand, "nextToken", "nextToken", "");

// src/pagination/ListCollectionsPaginator.ts

var paginateListCollections = (0, import_core.createPaginator)(OpenSearchServerlessClient, ListCollectionsCommand, "nextToken", "nextToken", "");

// src/pagination/ListLifecyclePoliciesPaginator.ts

var paginateListLifecyclePolicies = (0, import_core.createPaginator)(OpenSearchServerlessClient, ListLifecyclePoliciesCommand, "nextToken", "nextToken", "");

// src/pagination/ListSecurityConfigsPaginator.ts

var paginateListSecurityConfigs = (0, import_core.createPaginator)(OpenSearchServerlessClient, ListSecurityConfigsCommand, "nextToken", "nextToken", "");

// src/pagination/ListSecurityPoliciesPaginator.ts

var paginateListSecurityPolicies = (0, import_core.createPaginator)(OpenSearchServerlessClient, ListSecurityPoliciesCommand, "nextToken", "nextToken", "");

// src/pagination/ListVpcEndpointsPaginator.ts

var paginateListVpcEndpoints = (0, import_core.createPaginator)(OpenSearchServerlessClient, ListVpcEndpointsCommand, "nextToken", "nextToken", "");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  OpenSearchServerlessServiceException,
  __Client,
  OpenSearchServerlessClient,
  OpenSearchServerless,
  $Command,
  BatchGetCollectionCommand,
  BatchGetEffectiveLifecyclePolicyCommand,
  BatchGetLifecyclePolicyCommand,
  BatchGetVpcEndpointCommand,
  CreateAccessPolicyCommand,
  CreateCollectionCommand,
  CreateIndexCommand,
  CreateLifecyclePolicyCommand,
  CreateSecurityConfigCommand,
  CreateSecurityPolicyCommand,
  CreateVpcEndpointCommand,
  DeleteAccessPolicyCommand,
  DeleteCollectionCommand,
  DeleteIndexCommand,
  DeleteLifecyclePolicyCommand,
  DeleteSecurityConfigCommand,
  DeleteSecurityPolicyCommand,
  DeleteVpcEndpointCommand,
  GetAccessPolicyCommand,
  GetAccountSettingsCommand,
  GetIndexCommand,
  GetPoliciesStatsCommand,
  GetSecurityConfigCommand,
  GetSecurityPolicyCommand,
  ListAccessPoliciesCommand,
  ListCollectionsCommand,
  ListLifecyclePoliciesCommand,
  ListSecurityConfigsCommand,
  ListSecurityPoliciesCommand,
  ListTagsForResourceCommand,
  ListVpcEndpointsCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAccessPolicyCommand,
  UpdateAccountSettingsCommand,
  UpdateCollectionCommand,
  UpdateIndexCommand,
  UpdateLifecyclePolicyCommand,
  UpdateSecurityConfigCommand,
  UpdateSecurityPolicyCommand,
  UpdateVpcEndpointCommand,
  paginateListAccessPolicies,
  paginateListCollections,
  paginateListLifecyclePolicies,
  paginateListSecurityConfigs,
  paginateListSecurityPolicies,
  paginateListVpcEndpoints,
  AccessPolicyType,
  ConflictException,
  InternalServerException,
  ServiceQuotaExceededException,
  ValidationException,
  ResourceNotFoundException,
  StandbyReplicas,
  CollectionStatus,
  CollectionType,
  LifecyclePolicyType,
  ResourceType,
  VpcEndpointStatus,
  OcuLimitExceededException,
  IamIdentityCenterGroupAttribute,
  IamIdentityCenterUserAttribute,
  SecurityConfigType,
  SecurityPolicyType
});
